docroot/modules/custom/foia_file/drush/foia_file.drush.inc

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/**
+ * @file
+ * FOIA Api module drush command.
+ */
+
+use Drupal\file_entity\Entity\FileEntity;
+
+/**
+ * Implements hook_drush_command().
+ */
+function foia_file_drush_command() {
+  $items = [];
+
+  $items['file-entity-update'] = [
+    'description' => 'Updates file entities virus scan statuses.',
+    'core' => ['8+'],
+    'arguments' => [
+      'scan_result' => 'The result of the file scan.',
+    ],
+    'aliases' => ['feu'],
+  ];
+
+  return $items;
+}
+
+/**
+ * Implements hook_drush_help().
+ */
+function foia_file_drush_help($command) {
+  switch ($command) {
+    case 'drush:file-entity-update':
+      return dt('Update virus scan statuses for file entites based upon output from virus scanner.');
+  }
+}
+
+/**
+ * Implements drush_hook_COMMAND_validate().
+ */
+function drush_foia_file_file_entity_update_validate($scanOutput) {
+
+  $startTime = microtime(TRUE);
+
+  // Get allowed extensions.
+  $config = \Drupal::config('file_entity.settings');
+  $extensions = $config->get('default_allowed_extensions');
+  $pipeDelimitedExtensions = str_replace(" ", "|", $extensions);
+  $scans = explode("\n", $scanOutput);
+
+  foreach ($scans as $scan) {
+    if (!isset($scan) || is_null($scan)) {
+      $msg = 'A blank line was output by the virus scanner.';
+      $msg .= 'This is not expected behavior.';
+      \Drupal::logger('foia_file')->warning($msg);
+    }
+
+    // Split the scan line into filename and status.
+    preg_match('/[^ ]*$/', $scan, $wordsArray);
+    // Set status to the last word of the line.
+    $status = $wordsArray[0];
+
+    $validStates = ['OK', 'Removed.', 'FOUND'];
```

Comment:
Maybe we should rename this to `$validStatuses` to align with our usage of the `$status` variable.